wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -144,19 +149,27 @@ public void configurePipeline(PipelineConfigurer configurer) {
 
     try {
       Schema iSchema = configurer.getStageConfigurer().getInputSchema();
-      if (!config.containsMacro(Config.NAME_FIELD) && !(config.field.equals("*") || config.field.equals("#"))) {
+      if (!config.containsMacro(Config.NAME_FIELD) && !(config.getField().equals("*")
+        || config.getField().equals("#"))) {
         validateInputSchema(iSchema, collector);
       }
 
-      String directives = config.directives;
-      if (config.udds != null && !config.udds.trim().isEmpty()) {
+      String directives = config.getDirectives();
+      if (config.getUDDs() != null && !config.getUDDs().trim().isEmpty()) {
         if (config.containsMacro("directives")) {
-          directives = String.format("#pragma load-directives %s;", config.udds);
+          directives = String.format("#pragma load-directives %s;", config.getUDDs());
         } else {
-          directives = String.format("#pragma load-directives %s;%s", config.udds, config.directives);
+          directives = String.format("#pragma load-directives %s;%s", config.getUDDs(), config.getDirectives());
         }
       }
 
+      if (PRECONDITION_LANGUAGE_SQL.equalsIgnoreCase(config.getPreconditionLanguage())) {
```

Comment:
since the field can be macro, we should first check `if (!config.containsMacro(Config.NAME_PRECONDITION_LANGUAGE)`